File: app/src/main/java/com/abaffym/moviesapp/data/remote/rest/AuthorizationInterceptor.java
package com.abaffym.moviesapp.data.remote.rest;

import android.support.annotation.NonNull;

import java.io.IOException;

import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class AuthorizationInterceptor implements Interceptor {

    private static final String API_KEY_PARAM = "api_key";
    private static final String API_KEY = "c3e426c048101935674497dcc30cf7b7";

    @NonNull
    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {
        Request originalRequest = chain.request();

        HttpUrl originalHttpUrl = originalRequest.url();
        HttpUrl newHttpUrl = originalHttpUrl.newBuilder()
                .setQueryParameter(API_KEY_PARAM, API_KEY)
                .build();

        Request newRequest = originalRequest.newBuilder()
                .url(newHttpUrl)
                .build();

        return chain.proceed(newRequest);
    }
}

File: app/src/main/java/com/abaffym/moviesapp/data/remote/rest/AuthorizationInterceptor.kt
package com.abaffym.moviesapp.data.remote.rest

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

private const val API_KEY_PARAM = "api_key"
private const val API_KEY = "c3e426c048101935674497dcc30cf7b7"

class AuthorizationInterceptor : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val originalRequest = chain.request()

        val originalHttpUrl = originalRequest.url()
        val newHttpUrl = originalHttpUrl.newBuilder()
                .setQueryParameter(API_KEY_PARAM, API_KEY)
                .build()

        val newRequest = originalRequest.newBuilder()
                .url(newHttpUrl)
                .build()

        return chain.proceed(newRequest)
    }
}
